src/test/java/de/tk/opensource/secon/LdapConnectionTest.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright © 2020 Techniker Krankenkasse
+ * Copyright © 2020 BITMARCK Service GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package de.tk.opensource.secon;
+
+import global.namespace.fun.io.api.function.XFunction;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.test.StepVerifier;
+
+import java.net.URI;
+import java.security.cert.X509Certificate;
+import java.time.Duration;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.function.Function;
+
+import static de.tk.opensource.secon.SECON.directory;
+import static java.util.Locale.ENGLISH;
+import static org.junit.jupiter.api.Assertions.assertTimeoutPreemptively;
+import static reactor.core.scheduler.Schedulers.parallel;
+
+@Tag("LDAP")
+public class LdapConnectionTest {
```

Comment:
What exactly does this test do? And do we really need the reactor framework dependency?

Reply:
The test reveals the NameNotFoundException plus it simulates load on the LDAP server. The Reactor library is a test-only dependency which makes it easy to do the load testing.